### DISCUSSION TITLE
Support STATE_STANDBY for media_players

### DISCUSSION THREAD

**Discussion by ktnrg45** [home-assistant/architecture#504] [Entity Models] [answered]
## Context
These media_player integrations currently use STATE_STANDBY:

- androidtv
- appletv
- mediaroom
- ps4
- roku

Main issue is that STATE_STANDBY is mishandled when the PS4 integration is used in homekit and universal. STATE_STANDBY is interpreted as being not 'off'.
See home-assistant/core#28891

Also turning on from frontend card is mishandled.

## Proposal
Add handling of STATE_STANDBY in integrations:

- media_player
- homekit
- universal

Add/change handling of states in frontend media_player card.

## Consequences
Basic use of universal and homekit integrations and frontend card will be restored.

Logic using STATE_OFF and STATE_STANDBY may conflict.

**Comment by fleXible** [comment]
@ktnrg45 sorry for taking so long to come back to you.

I gave your proposal some thoughts and I'm kind of split.

On the one hand, this change would fix quite some issues without introducing breaking changes.

On the other hand, I don't see any good reason to introduce a second state, that essentially has the same meaning and function as STATE_OFF. It's also make the code more complicated and there are more line to change, than turning the proposal on its head and remove STATE_STANDBY from the few known places. On the downside, that would be a breaking change, but the state for "device is off" would be STATE_OFF instead of depending on the implementation. Documentation in this regard is also rather scarce or non existant, so you user first has to figure that out.

**Comment by frenck** [comment]
Close PR: <https://github.com/home-assistant/core/pull/30344>

It implemented the feature under discussion here.

**Comment by frenck** [comment]
I think this proposal/issue presented, can go multiple ways.

- This might be solved in the frontend, handling standby the same as on/off state.
- Solved by adjusting the entity model, making the `is_on` also respond false on standby.
- Refactor the entity model as a whole, implementing properties like `is_on`, so integrations can decide themselves (which is how it is implemented in most other entity models (e.g., Light, Switch).

There are multiple discussions and issues open around the entity model of the Media Player. And the thing is; it needs love. I guess what needs to happen first: is refactoring on a higher level.

As for the problem specifically presented: The PS4, maybe publish the off state instead of the standby state.

**Reply by fleXible** [reply]
I had a look into this way back then and came to the conclusion that MediaPlayerEntity should be derived of ToggleEntity instead of Entity and implement a default is_on method treating STATE_OFF as off and everything else as on.
Then components that use additional states may have their own implementation of this method.

It would bring media_player in line with how most models are implemented.

**Reply by pascalwinters** [reply]
I think it’s good to put the responsibility in the component. The component has the most knowledge of the meaning of the status.
So the PS4 can publish the off state when it’s actually in standby state.

**Reply by fleXible** [reply]
Personally I think STATE_STANDBY should never have been introduced into media_player. There is zero benefit to have it, but suddenly just a lot of weird corner cases introduced, that need special handling.
One example with PS4 and automation I had to tackle was, STATE_STANDBY doesn't complete replace STATE_OFF as one would think. And because there are several states for when the device is on, it just becomes a pain to differentiate on/off.

One example:
`{{ trigger.from_state.state in ("off", "standby") and is_state("media_player.pioneer", "off") }}`

**Reply by pascalwinters** [reply]
Yes I agree, for the PS4 case is off and standby both STATE_OFF. It makes the automations much easier and the media_player is also working correctly with the on/off toggle.

**Reply by pascalwinters** [reply]
How can we come to a decision so we can for example continue with the PS4 issue?

**Comment by bdraco** [comment]
HomeKit treats the following states as `off`
```
MEDIA_PLAYER_OFF_STATES = (
    STATE_OFF,
    STATE_UNKNOWN,
    STATE_STANDBY,
    "None",
)
```

**Reply by pascalwinters** [reply]
So for HomeKit is also a separate STATE_STANDBY is not necessary?

**Reply by fleXible** [reply]
No, STATE_OFF and STATE_STANDBY are both regarded as off states by Homekit.

**Comment by pascalwinters** [comment]
It's been relatively quiet on this topic for a while. How can we come to a conclusion, so we can continue with related issues/fixes?

**Comment by pascalwinters** [comment]
@fleXible / @frenck  How can we work towards a solution / decision?

**Reply by fleXible** [reply]
My point is quite clear, I'm for abandoning STATE_STANDBY as it only complicates things but doesn't add any value.

**Reply by pascalwinters** [reply]
@frenck What's your opinion?

**Reply by anubisg1** [reply]
Amy way we can get a decision? The issue is very annoying at the user level in so many ways

**Comment by pascalwinters** [comment]
Trying to come to a final decision.

IMHO, a separate STATE_STANDBY is not necessary so the off state can be published instead of the standby state.

**Comment by frenck** [comment, marked as answer]
We have re-iterated on this one, and concluded that for now we don't see enough traction for this.

Closing this, we can always decide to reconsider in the future.

../Frenck
